csr/csr.go

```diff
@@ -105,7 +105,10 @@ type names struct {
 // Alternative Names from the CSR. It guarantees that the SANs include the CN.
 // It also enforces various conditions on the CN, based on feature flags.
 func NamesFromCSR(csr *x509.CertificateRequest) names {
-	sans := csr.DNSNames
+	// Produce a new "sans" slice with the same memory address as csr.DNSNames
+	// but force a new allocation IFF an append happens so that we don't
+	// accidently mutate the underlying csr.DNSNames array.
```

Comment:
```suggestion
	// but force a new allocation if an append happens so that we don't
	// accidentally mutate the underlying csr.DNSNames array.
```

(if-and-only-if is cool but (a) good to spell it out when you need it and (b) I don't think we really need to state it here)